### ISSUE TITLE
[Bug]: Playwright tests fail to detect Next.js local server running with the `--experimental-https` flag.

### ISSUE THREAD

**Issue by manacy-keyvalue** [microsoft/playwright#29682]
### Version

1.41.2

### Steps to reproduce

1. Create a Next.js project.
2. Install Playwright: `npm install --save-dev @playwright/test`
3. In `playwright.config.ts` (or equivalent), configure webServer to use the development server with HTTPS:

```typescript
import { defineConfig } from '@playwright/test'

export default defineConfig({
  webServer: {
    command: 'next dev --experimental-https',
    url: 'https://localhost:3000',  // Adjust port if necessary
  },
  // ... other test configurations
})
```

4. Run Playwright tests: `npx playwright test`

### Expected behavior

- Playwright tests should launch a browser (in headed mode) and navigate to the running Next.js instance using the provided URL (Eg: `https://localhost:3000`).

### Actual behavior

Playwright tests fail to find the local server, resulting in test failures.

### Additional context

**Observations:**

- Playwright logs indicate it cannot find the server at the specified URL.
Terminal output:

playwright test --headed
[WebServer]  ⚠ Self-signed certificates are currently an experimental feature, use at your own risk.
[WebServer]  ⚠ Port 3000 is in use, trying 3001 instead.

- Browsers launched by Playwright remain blank or fail to load content.

**Additional Information:**

- Playwright version: 1.41.2
- Next.js version: 14.0.4

**Possible Causes:**

- Playwright's handling of self-signed certificates generated by Next.js with the `--experimental-https` flag

**Request:**

- Please investigate the issue and provide guidance on how to run Playwright tests against a Next.js server using the `--experimental-https` flag.
- If feasible, suggest alternative approaches to achieve secure testing with Playwright and Next.js.

### Environment

```shell
System:
    OS: macOS 13.4.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 280.95 MB / 16.00 GB
  Binaries:
    Node: 20.11.0 - /usr/local/bin/node
    npm: 10.4.0 - /usr/local/bin/npm
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: ^1.41.2 => 1.41.2
```

**Comment by pavelfeldman** [comment]
> [WebServer] ⚠ Port 3000 is in use, trying 3001 instead.

I think this is the key. You tell playwright to connect to 3000, but you run server on 3001.

**Comment by manacy-keyvalue** [comment]
I found the solution to the issue. Sharing it incase someone encounters similar issue in the future.

The crucial change was adding `ignoreHTTPSErrors: true` in the playwright config file under webServer configurations. Additionally, I used the -H flag when running the next dev command with the new hostname alias specified. A different alias was required for localhost in our use-case in-order to solve a project specific issue.

Elaborate details:
Change in the webServer configuration in the playwright config file:

`
  webServer: {
    command: 'npm run dev',
    url: baseURL,
    timeout: 120 * 1000,
    reuseExistingServer: true,
    ignoreHTTPSErrors: true,
  },
`

Usage of `-H` flag while running next dev command with the new hostname:
package.json file:
`
"scripts": {
    "dev": "next dev --experimental-https -H [hostname_alias]",
...
`
